tests/components/config/test_auth_provider_homeassistant.py

```diff
@@ -336,15 +336,35 @@ async def test_admin_change_password_not_owner(
     await auth_provider.async_validate_login("test-user", "test-pass")
 
 
-async def test_admin_change_password_no_creds(hass, hass_ws_client, owner_access_token):
-    """Test that change password fails with unknown credentials."""
+async def test_admin_change_password_no_user(hass, hass_ws_client, owner_access_token):
+    """Test that change password fails with unknown user."""
     client = await hass_ws_client(hass, owner_access_token)
 
     await client.send_json(
         {
             "id": 6,
             "type": "config/auth_provider/homeassistant/admin_change_password",
-            "username": "non-existing",
+            "user_id": "non-existing",
+            "password": "new-pass",
+        }
+    )
+
+    result = await client.receive_json()
+    assert not result["success"], result
+    assert result["error"]["code"] == "user_not_found"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6072450)